rc_simulator/radiation.py

```diff
@@ -28,28 +28,66 @@
 class Location(object):
     """Set the Location of the Simulation with an Energy Plus Weather File"""
 
-    def __init__(self, epwfile_path):
-
-        # Set EPW Labels and import epw file
-        epw_labels = ['year', 'month', 'day', 'hour', 'minute', 'datasource', 'drybulb_C', 'dewpoint_C', 'relhum_percent',
-                      'atmos_Pa', 'exthorrad_Whm2', 'extdirrad_Whm2', 'horirsky_Whm2', 'glohorrad_Whm2',
-                      'dirnorrad_Whm2', 'difhorrad_Whm2', 'glohorillum_lux', 'dirnorillum_lux', 'difhorillum_lux',
-                      'zenlum_lux', 'winddir_deg', 'windspd_ms', 'totskycvr_tenths', 'opaqskycvr_tenths', 'visibility_km',
-                      'ceiling_hgt_m', 'presweathobs', 'presweathcodes', 'precip_wtr_mm', 'aerosol_opt_thousandths',
-                      'snowdepth_cm', 'days_last_snow', 'Albedo', 'liq_precip_depth_mm', 'liq_precip_rate_Hour']
-
-        # Import EPW file
-        self.weather_data = pd.read_csv(
-            epwfile_path, skiprows=8, header=None, names=epw_labels).drop('datasource', axis=1)
+    # the location and name should be parsible from the EPW file
+    latitude_deg = None
+    longitude_deg = None
+    name = None
+
+    def __init__(self, path, dtype='epw'):
+
+        if dtype == 'epw':  # ewp weather data file
+
+            # Set EPW Labels and import epw file
+            epw_labels = ['year', 'month', 'day', 'hour', 'minute', 'datasource', 'drybulb_C', 'dewpoint_C',
+                          'relhum_percent', 'atmos_Pa', 'exthorrad_Whm2', 'extdirrad_Whm2', 'horirsky_Whm2',
+                          'glohorrad_Whm2', 'dirnorrad_Whm2', 'difhorrad_Whm2', 'glohorillum_lux',
+                          'dirnorillum_lux', 'difhorillum_lux', 'zenlum_lux', 'winddir_deg', 'windspd_ms',
+                          'totskycvr_tenths', 'opaqskycvr_tenths', 'visibility_km', 'ceiling_hgt_m',
+                          'presweathobs', 'presweathcodes', 'precip_wtr_mm', 'aerosol_opt_thousandths',
+                          'snowdepth_cm', 'days_last_snow', 'Albedo', 'liq_precip_depth_mm',
+                          'liq_precip_rate_Hour']
+
+            # Import EPW file
+            self.weather_data = pd.read_csv(path, skiprows=8, header=None, names=epw_labels).drop('datasource', axis=1)
+
+            # extract the lat/lon from the EPW header
+            fin = open(path,'r')
+            for line in fin:
+                if 'LOCATION' in line:
+                    args = line.split(',')
+
+                    try:
+                        self.name = args[1]
+                        self.latitude_deg = float(args[6])
+                        self.longitude_deg = float(args[7])
+                    except:
+                        pass
+                    break
+
+            if self.latitude_deg==None or self.longitude_deg==None:
+                # trap problem with 
+                print("ERROR: EPW weather file contains incomplete",
```

Comment:
could you replace the print statement with an Raise Value Error

`raise ValueError("EPW weather file contains an incomplete header please see ....")`

See 8.4 https://docs.python.org/2/tutorial/errors.html